frontend/src/component/pages/shared/Header.tsx

```diff
@@ -1,12 +1,20 @@
 import React, { Component } from 'react';
+import { Button } from '../../ui/Button';
 import './Header.scss';
 
-export class Header extends Component {
+interface Props {
+  onSignOutButtonClick: () => void;
+}
+
+export class Header extends Component<Props> {
   render() {
     return (
       <header>
         <div className={'center'}>
```

Comment:
Let's open an issue and add `<!-- TODO(issue#your_issue_number): use flex layout to prepare for search box with dynamic width. -->` before those lines.

You can read about flex layout here!!!: https://css-tricks.com/snippets/css/a-guide-to-flexbox

Reply:
Discussed offline. This is not needed anymore.